Sage/carbon
# feat(click-away-wrapper): add ClickAwayWrapper component to internal directory

### Proposed behaviour


Adds `ClickAwayWrapper` component to handle triggering an action if a mousedown/ click event is detected
outside of it

Adds the `ClickAwayWrapper` to enable the `PopoverContainer` to close when a `mousedown` event is
detected from outside of the wrapper boundaries

Adds `ClickAwayWrapper` to `Date` and `Submenu` as well

fix #5017, fix #5158

### Current behaviour


No `ClickAwayWrapper` component exists. `PopoverContainer` does not close when click is outside of it. `Date` and `Submenu` have their own implementations to handle closing when click outside occurs

### Checklist




#### QA


### Additional context



### Testing instructions


Test Date, Submenu and PopoverContainer stories, clicking inside should not close the components but clicking outside should

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


https://codesandbox.io/s/mystifying-northcutt-c3slyx?file=/src/index.js
